node_modules/col-activities/lib/notifications/daily.js

```diff
@@ -0,0 +1,548 @@
+      // so we simple get the basic user profiles
+      {'model': DB.User, 'as': 'users', 'required': true, 'attributes': UserConstants.BASIC_USER_FIELDS},
+
+      // The comments
+      {'model': DB.Comment, 'required': true, 'where': {'created_at': {'gt': since}}, 'include': [
+        {'model': DB.Comment, 'required': false, 'as': 'parent'}
+      ]}
+    ],
+    'limit': null,
+    'subQuery': false
+  };
+  DB.Asset.findAll(options).complete(function(err, assets) {
+    if (err) {
+      log.error({'err': err, 'course': ctx.course.id}, 'Unable to get the assets with recent comments for a course');
+      return callback(err);
+    }
+
+    assets = _.map(assets, function(asset) {
+      asset = asset.toJSON();
+
+      // Index the asset owners by their ID for quick lookups
+      asset.users = _.indexBy(asset.users, 'id');
+
+      // Add the user objects for the commenters
+      asset.comments = _.map(asset.comments, function(comment) {
+        comment = comment.toJSON();
+
+        comment.user = users[comment.user_id];
+        if (comment.parent) {
+          comment.parent.user = users[comment.parent.user_id];
+        }
+
+        return comment;
+      });
+
+      return asset;
+    });
+
+    return callback(null, assets);
+  });
+};
+
+/**
+ * Get the assets with recent chat messages
+ *
+ * @param  {Context}    ctx                     Standard context containing the current user and the current course
+ * @param  {Object}     users                   The users in the course (keyed by their user id)
+ * @param  {String}     since                   The timestamp (`YYYY-MM-DD HH:mm:ss`) after which the chat messages have to take place
+ * @param  {Function}   callback                Standard callback function
+ * @param  {Object}     callback.err            An error object, if any
+ * @param  {Object}     callback.whiteboards    Whiteboards with recent chat messages
+ * @api private
+ */
+var getCourseWhiteboardData = function(ctx, users, since, callback) {
+  var queryOptions = {
+    'where': {
+      'course_id': ctx.course.id
+    },
+    'include': [
+      {
+        'model': DB.User
+      },
+      {
+        'model': DB.Chat,
+        'where': {
+          'created_at': {
+            'gt': since
+          }
+        }
+      }
+    ]
+  };
+  DB.Whiteboard.findAll(queryOptions).complete(function(err, whiteboards) {
+    if (err) {
+      log.error({'err': err, 'course': ctx.course.id}, 'Unable to get the whiteboards with recent chat messages for a course');
+      return callback(err);
+    }
+
+    // Add the user object of the chatter to each chat message
+    whiteboards = _.map(whiteboards, function(whiteboard) {
+      whiteboard = whiteboard.toJSON();
+
+      whiteboard.chats = _.chain(whiteboard.chats)
+        .map(function(chat) {
+          chat.user = users[chat.user_id];
+          return chat;
+        })
+        .sortBy('id')
```

Comment:
Can we not specify that the chat messages should be ordered by id in `queryOptions`